source/common/router/router.cc

```diff
@@ -1481,6 +1503,12 @@ void Filter::UpstreamRequest::onResetStream(Http::StreamResetReason reason,
                                             absl::string_view transport_failure_reason) {
   ScopeTrackerScopeState scope(&parent_.callbacks_->scope(), parent_.callbacks_->dispatcher());
 
+  if (span_ != nullptr) {
+    // Add tags about reset.
+    span_->setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
```

Comment:
do we need to add both Error and ErrorReason? Maybe users could infer Error or not from ErrorReason being non-empty?

otherwise lgtm

Reply:
`error` is a standard tag, so most tracing tools would expect to see this to highlight them in a UI.

One option, if there was the concept of a "trace level" parameter, would be to only generate the reason field if a debug level had been set.

Reply:
I don't think having an extra reason field would change performance that much. But I can follow this approach if needed.